openedx/core/djangoapps/programs/views.py

```diff
@@ -59,10 +59,13 @@ def get_course_programs_for_dashboard(user, course_keys):   # pylint: disable=in
     # reindex the result from pgm -> course code -> course run
     #  to
     # course run -> program, ignoring course runs not present in the dashboard enrollments
-    for program in programs:
-        for course_code in program['course_codes']:
-            for run in course_code['run_modes']:
-                if run['course_key'] in course_keys:
-                    course_programs[run['course_key']] = program
+    try:
+        for program in programs:
+            for course_code in program['course_codes']:
+                for run in course_code['run_modes']:
+                    if run['course_key'] in course_keys:
+                        course_programs[run['course_key']] = program
+    except KeyError:
+        log.exception('Error parsing the Programs API response.')
```

Comment:
i think we should move the try / except inside of the loop, around each program, so that encountering one broken program out of many wouldn't cause all of them to be hidden.

also, please include the data of the program that raised the `KeyError` in the log message, using `%r`